bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/PdfViewerImpl.java

```diff
@@ -0,0 +1,210 @@
+
+    @PostConstruct
+    protected void initModel() {
+        ConfigurationBuilder cb = resource.adaptTo(ConfigurationBuilder.class);
+        if (cb != null) {
+            caConfig = cb.as(PdfViewerCaConfig.class);
+        }
+    }
+
+    @Override
+    public String getClientId() {
+        return caConfig.clientId();
+    }
+
+    @Override
+    public String getReportSuiteId() {
+        return caConfig.reportSuiteId();
+    }
+
+    @Override
+    public String getDocumentPath() {
+        return documentPath;
+    }
+
+    @Override 
+    public String getDocumentFileName() {
+        int index = documentPath.lastIndexOf("/");
+        return documentPath.substring(index + 1);
+    }
+
+    @Override
+    public String getType() {
+        return type;
+    }
+
+    @Override
+    public String getDefaultViewMode() {
+        return defaultViewMode;
+    }
+
+    @Override
+    public boolean isBorderless() {
+        return borderless;
+    }
+
+    @Override
+    public boolean isShowAnnotationTools() {
+        return showAnnotationTools;
+    }
+
+    @Override
+    public boolean isShowFullScreen() {
+        return showFullScreen;
+    }
+
+    @Override
+    public boolean isShowLeftHandPanel() {
+        return showLeftHandPanel;
+    }
+
+    @Override
+    public boolean isShowDownloadPdf() {
+        return showDownloadPdf;
+    }
+
+    @Override
+    public boolean isShowPrintPdf() {
+        return showPrintPdf;
+    }
+
+    @Override
+    public boolean isShowPageControls() {
+        return showPageControls;
+    }
+
+    @Override
+    public boolean isDockPageControls() {
+        return dockPageControls;
+    }
+
+    @Override
+    public String getViewerConfigJson() {
+        JsonObjectBuilder jsonObjectBuilder = Json.createObjectBuilder();
+        jsonObjectBuilder.add(FIELD_EMBED_MODE, type);
+
+        if(!StringUtils.isEmpty(type)) {
+            if (type.equals(FULL_WINDOW)) {
+                jsonObjectBuilder.add(PN_DEFAULT_VIEW_MODE, defaultViewMode);
+                jsonObjectBuilder.add(PN_SHOW_ANNOTATION_TOOLS, showAnnotationTools);
+                jsonObjectBuilder.add(PN_SHOW_LEFT_HAND_PANEL, showLeftHandPanel);
+            }
+
+            if (type.equals(SIZED_CONTAINER)) {
+                jsonObjectBuilder.add(PN_SHOW_FULL_SCREEN, showFullScreen);
+            }
+
+            if (type.equals(FULL_WINDOW) || type.equals(SIZED_CONTAINER)) {
+                jsonObjectBuilder.add(PN_SHOW_PAGE_CONTROLS, showPageControls);
+                jsonObjectBuilder.add(PN_DOCK_PAGE_CONTROLS, dockPageControls);
+            }
+        }
+
+        jsonObjectBuilder.add(PN_SHOW_DOWNLOAD_PDF, showDownloadPdf);
+        jsonObjectBuilder.add(PN_SHOW_PRINT_PDF, showPrintPdf);
+
+        return jsonObjectBuilder.build().toString();
+    }
+
+    @Override
+    public String getContainerClass() {
+        if(!StringUtils.isEmpty(type)) {
+            if (type.equals(FULL_WINDOW) && borderless) {
```

Comment:
I'd use StringUtils.equals() here and below.
`StringUtils.isEmpty(type)` won't be needed anymore.

Reply:
Done in a691b53